### PR TITLE
Fix setup callback hook to pass LightningModule through

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

As currently written, the `setup()` function in callbacks is passed `model=None` since `Trainer.model` is not yet set when the call to `self.trainer.get_model()` is made. This PR explicitly passes model through to the callback instead of relying on `self.trainer.get_model()`

cc @kandluis

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by ananthsub** [comment]
cc @awaelchli @SeanNaren @tchaton

**Review by SeanNaren — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Great catch !

**Review by ananyahjha93 — APPROVED** [review]

**Review by Borda — COMMENTED** [review]
LGTM, just it seems to be API change for setup hook

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -850,7 +850,7 @@ def call_setup_hook(self, model):
             called = self.datamodule.has_setup_test if self.testing else self.datamodule.has_setup_fit
             if not called:
                 self.datamodule.setup(stage_name)
-        self.setup(stage_name)
+        self.setup(model, stage_name)
```
Comment (ananthsub) [inline pytorch_lightning/trainer/trainer.py:853]: @Borda @tchaton  one thing I need to double check: if `setup` is called after DDP. do we need to unwrap `model` here? 

another fix would be be setting `Trainer.model` before `setup` is called. Which do you think is better?

**Comment by Borda** [comment]
@williamFalcon could you pls approve this API/hook change as part of bugfix release?

File: pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (williamFalcon) [reply pytorch_lightning/trainer/trainer.py:853]: we're good here. setup gets called before wrapping the model in ddp

**Pull request merged**